pkg/rc/replication_controller.go

```diff
@@ -316,6 +318,70 @@ func (rc *replicationController) addPods(current types.PodLocations, eligible []
 	return nil
 }
 
+func (rc *replicationController) allocateNodes(currentNodes []types.NodeName, eligibleNodes []types.NodeName) error {
+	var err error
+	if rc.AllocationStrategy == fields.CattleStrategy {
+		err = rc.allocateCattleNodes(currentNodes, eligibleNodes)
+	} else {
+		err = rc.allocatePetNodes(currentNodes, eligibleNodes)
+	}
+
+	if err != nil {
+		return util.Errorf("Unable to allocate nodes: %s", err)
+	}
+
+	return nil
+}
+
+func (rc *replicationController) allocateCattleNodes(currentNodes []types.NodeName, eligibleNodes []types.NodeName) error {
+	rcStatus, _, err := rc.rcStatusStore.Get(rc.ID())
+	isNoNodeTransferInProgress := (err != nil && statusstore.IsNoStatus(err)) ||
+		(err == nil && rcStatus.NodeTransfer == nil)
+	if isNoNodeTransferInProgress {
+		rc.mu.Lock()
+		newNodes, err := rc.scheduler.AllocateNodes(rc.Manifest, rc.NodeSelector, len(currentNodes)+1)
+		rc.mu.Unlock()
+		if err != nil || len(newNodes) < 1 {
+			errMsg := fmt.Sprintf(
+				"Unable to allocate nodes over grpc: %d replicas desired, %d currentNodes, %d eligible. Error: %s",
+				rc.ReplicasDesired, len(currentNodes), len(eligibleNodes), err,
+			)
+			err := rc.alerter.Alert(rc.alertInfo(errMsg))
+			if err != nil {
+				rc.logger.WithError(err).Errorln("Unable to send alert")
+			}
+
+			return util.Errorf(errMsg)
+		}
+
+		// Right now we do not support multiple node transfers, so we just
+		// write the first
+		rcStatus.NodeTransfer.NewNode = newNodes[0]
+		err = rc.rcStatusStore.Set(rc.ID(), rcStatus)
```

Comment:
I don't think it's strictly necessary but it'd probably be good practice if we used a transaction here. either a "check-and-set" using the modify index from the previous `Get()` call or do it in a full on transaction (would be a check-and-set either way)